Core/OfficeDevPnP.Core.Tests/Framework/ObjectHandlers/ObjectSiteSecurityTests.cs

```diff
@@ -274,15 +264,51 @@ public void CanCreateEntities2()
         {
             using (var ctx = TestCommon.CreateClientContext())
             {
+                Web web = ctx.Web;
+
                 // Load the base template which will be used for the comparison work
-                var creationInfo = new ProvisioningTemplateCreationInformation(ctx.Web) { BaseTemplate = ctx.Web.GetBaseTemplate() };
+                var creationInfo = new ProvisioningTemplateCreationInformation(web) { BaseTemplate = web.GetBaseTemplate() };
                 creationInfo.IncludeSiteGroups = true;
                 var template = new ProvisioningTemplate();
-                template = new ObjectSiteSecurity().ExtractObjects(ctx.Web, template, creationInfo);
+                template = new ObjectSiteSecurity().ExtractObjects(web, template, creationInfo);
+
+                ctx.Load(web,
+                    w => w.AssociatedOwnerGroup.Title,
+                    w => w.AssociatedMemberGroup.Title,
+                    w => w.AssociatedVisitorGroup.Title);
+                ctx.ExecuteQuery();
 
                 Assert.IsTrue(template.Security.AdditionalAdministrators.Any());
                 Assert.IsTrue(template.Security.SiteGroups.Any());
+                Assert.AreEqual(SiteTitleToken.GetReplaceToken(web.AssociatedOwnerGroup.Title, web), template.Security.AssociatedOwnerGroup, "Associated owner group title mismatch.");
```

Comment:
@jensotto Could this fail when the site used for testing is created right before this test? There is some asynchronous action going on that initially names the default associated groups like "Site Members" (without the title) and then some seconds (minutes?) after site creation changes the group names to include the site title.
Just want to be sure not to introduce timing issues here making the test fail sometimes and sometimes not.

Reply:
That should not matter as this is only a unit test. The unit tests runs against an already existing site collection, so this should not be an issue.

Reply:
Ok I'll try another angle. I'm concerned about code like the following and timing issues arising from the fact that SharePoint changes the group name behind our back several seconds after running this code:

https://github.com/SharePoint/PnP-Sites-Core/blob/93295420e78549a496936d71ec07655255fda2e0/Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectSiteSecurity.cs#L64

This could create a group although it is not necessary.

I just tested with a newly created site. On my developer tenant it takes 18 seconds (!) after site creation for the initial group names to change from e.g. "Site Owners" to "Site Title Owners". So code checking group names has a time window of 18 seconds where it sees the "old" names. And then suddenly the "new" names.

(Somewhere I saw a commend I think by @jansenbe referring to timing issues and site creation.)

My thesis is that we have to wait for the group names to be "final" before doing anything with them. So this would be an 18-second wait in my specific case. And my questions are: are my concerns reasonable? If yes: is this already being taken care about in the framework? Maybe @VesaJuvonen could chime in and give us a hint - do we have to care about this delay of several seconds between site creation and changing the group names?

Reply:
You know what? I created a test that checks what I'd expect. No need to speculate :) -> https://github.com/SharePoint/PnP-Sites-Core/pull/2192